### PR TITLE
Refactor: modernize code and remove unnecessary bits

### PR DESCRIPTION
**Pull request by eumiro**
This is just a refactoring consisting of a few independent steps:
1. one missing f-string, literal sets and a one forgotten python2 style class
2. removal of all unnecessary parentheses
3. fix docstrings format to triple quotes
4. replace `len([x for x in collection if cond(x)])` with `sum(cond(x) for x in collection)` to compact the code and avoid creating whole list

Please feel free to refuse/rollback any proposed changes.

### CODE REVIEW

File: examples/subclass/plot_antigraph.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ def adjacency(self):
         """
         nodes = set(self.adj)
         for n, nbrs in self.adj.items():
-            yield (n, nodes - set(nbrs) - {n})
+            yield n, nodes - set(nbrs) - {n}
```
Comment (rossbar) [inline examples/subclass/plot_antigraph.py:155]: This is entirely subjective, but I actually prefer the `()` as this makes it more immediately clear that this is a tuple (it's easy to miss the comma).

This is more of a style consideration though - I'm surprised `black` doesn't have an opinion on it.

**Review by rossbar — COMMENTED** [review]
I didn't look through everything in detail, but +1 for rm Python2-style class definitions, `"""` for starting docstrings, and clearer list-comps (I haven't checked every change in detail to ensure they're equivalent).

The majority of the changes revolve around removing parentheses from tuples. I'm ambivalent on this (in some cases I think keeping the parentheses makes the code more readable), but I'm +1 for consistency. We already use `black` for code style - could these changes be included there? These changes seem like they originated from some sort of automated linter - if we could decide on a tuple style and included it in our existing linting that would be an improvement.

**Comment by eumiro** [comment]
There's `pyupgrade` and there's code inspect in PyCharm. And every one of them have their own opinion.

The number 4 (`sum(cond for elem in iter)`) was manual because not everything can be converted that way. For instance if `cond` returns an integer that would be evaluated to `True/False` at the end after `if` then putting this condition to the beginning just after `sum(` would affect the sum in an unexpected way. That's one point where I'd like to hear your opinion.

**Comment by dschult** [comment]
The trouble with PRs like this is that they are very hard to review, hard to make sure they don't have alternate impact, and let's face it, they don't actually help the code much.

I am most leery of the list comp changes that sum over True/False to use the bool trait that addition is like the numbers 1 and 0. `sum(val == 5 for val in ttt)` is just not as readable as `sum(1 for val in ttt if val == 5)` or `len([val for val in ttt if val == 5])`.  The last could create a large list, but maybe it is known not to. An automated translate -- and even a large scale manual translate -- doesn't take the context fully into account.

I'm also -1 on removing parens from tuples that are return values -- and many other cases actually.  Let's let Black do the linting so we don't have to.  If black doesn't have a preference, then let's not worry about it -- or else bring up the issue with black.

Even removing python2 class syntax has almost no benefit to offset the noise in github's "blame" feature. But I suppose I could go along with that. Most of these are in tests and maybe we can wait until we convert a bunch of test classes into test functions -- which I see on the horizon... :}

**Comment by eumiro** [comment]
@dschult I totally agree with the `git blame` noise and lower added value of such commits. They should be scheduled together with major syntax updates such as f-strings, await/async keywords, or pathlib (and preferably together with `black` version updates that possibly reformat the whole codebase). Please feel free to close this PR, but I am glad this discussion happened.

**Pull request closed without merging**